.github/workflows/main.yml

```diff
@@ -0,0 +1,134 @@
+name: CI Pipeline
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'python-flask/**'
+      - 'c-sharp-razor/**'
+      - 'rust-actix/**'
+      - 'ruby-sinatra/**'
+      - 'javascript-express/**'
+      - 'go-gin/**'
+      - 'go-gorilla/**'
+      - '/**'
+
+jobs:
+  check-root-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      root_changed: ${{ steps.set-flags.outputs.root_changed }}
+      service_changed: ${{ steps.set-flags.outputs.service_changed }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Check if root files or any service changed
```

Comment:
**For this step I ended up using:
tj-actions/changed-files@v34: https://github.com/marketplace/actions/changed-files
and works great! I had a bunch of problems trying to figure out the command and ended up using this which works great. But if your version works leave it.

Reply:
Yes I saw it! I'll try and add it and see if it works just so we have a little consistency between the CI and CD :)

Reply:
Actually, I think I will put it in the backlog. The CI works as it should and I think we should not over-engineer it and spend too much time on devops stuff imo. :)) <3

Reply:
So if its okay, i will leave as it is.